lib/active_merchant/billing/gateways/authorize_net.rb

```diff
@@ -62,82 +62,50 @@ def initialize(options={})
       end
 
       def purchase(amount, payment, options = {})
-        commit("AUTH_CAPTURE") do |xml|
-          add_order_id(xml, options)
-          xml.transactionRequest do
-            xml.transactionType('authCaptureTransaction')
-            xml.amount(amount(amount))
-
-            add_payment_source(xml, payment)
-            add_invoice(xml, options)
-            add_customer_data(xml, payment, options)
-            add_market_type(xml, payment)
-            add_settings(xml, payment, options)
-            add_user_fields(xml, amount, options)
+        if payment.is_a?(String)
+          commit(:cim_purchase) do |xml|
+            add_cim_auth_purchase(xml, "profileTransAuthCapture", amount, payment, options)
+          end
+        else
+          commit(:purchase) do |xml|
+            add_auth_purchase(xml, "authCaptureTransaction", amount, payment, options)
           end
         end
       end
 
       def authorize(amount, payment, options={})
-        commit("AUTH_ONLY") do |xml|
-          add_order_id(xml, options)
-          xml.transactionRequest do
-            xml.transactionType('authOnlyTransaction')
-            xml.amount(amount(amount))
-
-            add_payment_source(xml, payment)
-            add_invoice(xml, options)
-            add_customer_data(xml, payment, options)
-            add_market_type(xml, payment)
-            add_settings(xml, payment, options)
-            add_user_fields(xml, amount, options)
+        if payment.is_a?(String)
+          commit(:cim_authorize) do |xml|
+            add_cim_auth_purchase(xml, "profileTransAuthOnly", amount, payment, options)
+          end
+        else
+          commit(:authorize) do |xml|
+            add_auth_purchase(xml, "authOnlyTransaction", amount, payment, options)
           end
         end
       end
 
       def capture(amount, authorization, options={})
-        commit("PRIOR_AUTH_CAPTURE") do |xml|
-          add_order_id(xml, options)
-          xml.transactionRequest do
-            xml.transactionType('priorAuthCaptureTransaction')
-            xml.amount(amount(amount))
-            xml.refTransId(split_authorization(authorization)[0])
-
-            add_invoice(xml, options)
-            add_user_fields(xml, amount, options)
-          end
+        if (auth_was_for_cim?(authorization))
```

Comment:
@masaruhoshi is there a particular style guide you're targeting? In general we haven't enforced a style guide on AM, so I don't think these tweaks are necessarily germane.